include/alpaka/math/MathGenericSycl.hpp

```diff
@@ -0,0 +1,549 @@
+        {
+            return sycl::cos(arg);
+        }
+    };
+
+    //! The SYCL erf trait specialization.
+    template<typename TArg>
+    struct Erf<experimental::math::ErfGenericSycl, TArg, std::enable_if_t<std::is_floating_point_v<TArg>>>
+    {
+        static auto erf(experimental::math::ErfGenericSycl const&, TArg const& arg)
+        {
+            return sycl::erf(arg);
+        }
+    };
+
+    //! The SYCL exp trait specialization.
+    template<typename TArg>
+    struct Exp<experimental::math::ExpGenericSycl, TArg, std::enable_if_t<std::is_floating_point_v<TArg>>>
+    {
+        static auto exp(experimental::math::ExpGenericSycl const&, TArg const& arg)
+        {
+            return sycl::exp(arg);
+        }
+    };
+
+    //! The SYCL floor trait specialization.
+    template<typename TArg>
+    struct Floor<experimental::math::FloorGenericSycl, TArg, std::enable_if_t<std::is_floating_point_v<TArg>>>
+    {
+        static auto floor(experimental::math::FloorGenericSycl const&, TArg const& arg)
+        {
+            return sycl::floor(arg);
+        }
+    };
+
+    //! The SYCL fmod trait specialization.
+    template<typename Tx, typename Ty>
+    struct Fmod<
+        experimental::math::FmodGenericSycl,
+        Tx,
+        Ty,
+        std::enable_if_t<std::is_floating_point_v<Tx> && std::is_floating_point_v<Ty>>>
+    {
+        static auto fmod(experimental::math::FmodGenericSycl const&, Tx const& x, Ty const& y)
+        {
+            return sycl::fmod(x, y);
+        }
+    };
+
+    //! The SYCL isfinite trait specialization.
+    template<typename TArg>
+    struct Isfinite<experimental::math::IsfiniteGenericSycl, TArg, std::enable_if_t<std::is_floating_point_v<TArg>>>
+    {
+        auto operator()(experimental::math::IsfiniteGenericSycl const&, TArg const& arg)
+        {
+            return sycl::isfinite(arg);
+        }
+    };
+
+    //! The SYCL isinf trait specialization.
+    template<typename TArg>
+    struct Isinf<experimental::math::IsinfGenericSycl, TArg, std::enable_if_t<std::is_floating_point_v<TArg>>>
+    {
+        auto operator()(experimental::math::IsinfGenericSycl const&, TArg const& arg)
+        {
+            return sycl::isinf(arg);
+        }
+    };
+
+    //! The SYCL isnan trait specialization.
+    template<typename TArg>
+    struct Isnan<experimental::math::IsnanGenericSycl, TArg, std::enable_if_t<std::is_floating_point_v<TArg>>>
+    {
+        auto operator()(experimental::math::IsnanGenericSycl const&, TArg const& arg)
+        {
+            return sycl::isnan(arg);
+        }
+    };
+
+    //! The SYCL log trait specialization.
+    template<typename TArg>
+    struct Log<experimental::math::LogGenericSycl, TArg, std::enable_if_t<std::is_floating_point_v<TArg>>>
+    {
+        static auto log(experimental::math::LogGenericSycl const&, TArg const& arg)
+        {
+            return sycl::log(arg);
+        }
+    };
```

Comment:
How can this work? It should be `operator()`:
```suggestion
    struct Log<experimental::math::LogGenericSycl, TArg, std::enable_if_t<std::is_floating_point_v<TArg>>>
    {
        auto operator()(experimental::math::LogGenericSycl const&, TArg const& arg)
        {
            return sycl::log(arg);
        }
    };
```
Applies to several other places as well.

Reply:
Looks like I missed a refactoring of the math functions here. Thanks for noticing.